compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -353,9 +356,12 @@ public void visit(BLangPackage pkgNode) {
             analyzeNode(classDefinition, pkgEnv);
         }
 
-        for (BLangFunction function : pkgNode.functions) {
-            analyzeNode(function, pkgEnv);
-        }
+        pkgNode.functions.forEach(function -> {
+            // Skip visiting worker lambdas.
```

Comment:
Because they are already visited? Shall we improve the comment and use a simple for loop?